File: frontend/src/lib/api/shellGift.ts
import apiClient from '@/lib/apiClient'

export interface ShellGiftItem {
  id: string
  task_id: string
  from_user_id: string
  to_user_id: string
  ai_title: string
  ai_content: string
  cover_url: string
  photo_url: string | null
  is_opened: boolean
  opened_at: string | null
  created_at: string
}

export function getShellGifts(): Promise<ShellGiftItem[]> {
  return apiClient.get('/api/v1/shell-gifts').then((r) => r.data)
}

export function openShellGift(id: string): Promise<ShellGiftItem> {
  return apiClient.post(`/api/v1/shell-gifts/${id}/open`).then((r) => r.data)
}

File: backend/internal/dto/shell_gift.go
package dto

import "time"

type ShellGiftItem struct {
	ID         string `json:"id"`
	TaskID     string `json:"task_id"`
	FromUserID string `json:"from_user_id"`
	ToUserID   string `json:"to_user_id"`

	AITitle   string `json:"ai_title"`
	AIContent string `json:"ai_content"`

	CoverURL string  `json:"cover_url"`
	PhotoURL *string `json:"photo_url"`

	IsOpened  bool       `json:"is_opened"`
	OpenedAt  *time.Time `json:"opened_at"`
	CreatedAt time.Time  `json:"created_at"`
}
